internal/topic/core_test.go

```diff
@@ -68,3 +72,472 @@ func TestCore_List(t *testing.T) {
+	ctx := context.Background()
+	dTopic := getDummyTopicModel()
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Test1",
+			fields: fields{
+				repo:        mockTopicRepo,
+				projectCore: mockProjectCore,
+				brokerStore: mockBrokerStore,
+			},
+			args: args{
+				ctx:   ctx,
+				model: dTopic,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
+				repo:        tt.fields.repo,
+				projectCore: tt.fields.projectCore,
+				brokerStore: tt.fields.brokerStore,
+			}
+			mockBrokerStore.EXPECT().GetAdmin(gomock.Any(), messagebroker.AdminClientOptions{}).Return(mockAdmin, nil)
+			mockAdmin.EXPECT().CreateTopic(gomock.Any(), messagebroker.CreateTopicRequest{dTopic.Name, DefaultNumPartitions}).Return(messagebroker.CreateTopicResponse{}, nil)
+			if err := c.CreateSubscriptionTopic(tt.args.ctx, tt.args.model); (err != nil) != tt.wantErr {
+				t.Errorf("Core.CreateSubscriptionTopic() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestCore_ExistsWithName(t *testing.T) {
+	type fields struct {
+		repo        IRepo
+		projectCore project.ICore
+		brokerStore brokerstore.IBrokerStore
+	}
+	type args struct {
+		ctx       context.Context
+		name      string
+		projectID string
+		topicName string
+	}
+	ctrl := gomock.NewController(t)
+	mockTopicRepo := topicrepomock.NewMockIRepo(ctrl)
+	mockProjectCore := projectcoremock.NewMockICore(ctrl)
+	mockBrokerStore := brokerstoremock.NewMockIBrokerStore(ctrl)
+	ctx := context.Background()
+	dTopic := getDummyTopicModel()
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "Check if Topic exists with name.",
+			fields: fields{
+				repo:        mockTopicRepo,
+				projectCore: mockProjectCore,
+				brokerStore: mockBrokerStore,
+			},
+			args: args{
+				ctx:       ctx,
+				name:      dTopic.Name,
+				projectID: dTopic.ExtractedProjectID,
+				topicName: dTopic.ExtractedTopicName,
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name: "Throw error for invalid topic name.",
+			fields: fields{
+				repo:        mockTopicRepo,
+				projectCore: mockProjectCore,
+				brokerStore: mockBrokerStore,
+			},
+			args: args{
+				ctx:       ctx,
+				name:      "",
+				projectID: "",
+				topicName: "",
+			},
+			want:    false,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
+				repo:        tt.fields.repo,
+				projectCore: tt.fields.projectCore,
+				brokerStore: tt.fields.brokerStore,
+			}
+			var err2 error = nil
+			var expectBool bool = true
+			if len(tt.args.name) == 0 {
+				err2 = fmt.Errorf("Invalid Project Name!")
+				expectBool = false
+			} else {
+				mockTopicRepo.EXPECT().Exists(gomock.Any(), common.GetBasePrefix()+Prefix+tt.args.projectID+"/"+tt.args.topicName).Return(expectBool, err2)
```

Comment:
Use `AssignableToTypeOf(context.Context)` instead of `gomock.Any()` for ctx values.